src/backend/concurrency/transaction_manager.h

```diff
@@ -55,6 +55,83 @@ class TransactionManager {
       const storage::TileGroupHeader *const tile_group_header,
       const oid_t &tuple_id) = 0;
 
+  bool VisibleTupleExist(std::vector<ItemPointer> & locations){
+    bool visible_tuple_exist = false;
+
+    if (locations.size() > 0) {
+      for(unsigned long i = 0; i < locations.size(); i++) {
+        auto tile_group_header = catalog::Manager::GetInstance()
+            .GetTileGroup(locations[i].block)->GetHeader();
+        auto tuple_id = locations[i].offset;
+        if (this->IsVisible(tile_group_header, tuple_id)) {
+          visible_tuple_exist = true;
+          break;
+        }
+      }
+    }
+
+    return visible_tuple_exist;
+  }
+
+  bool IsVisbleOrDirty(__attribute__((unused)) const storage::Tuple *key,
```

Comment:
Please remove this function. I have renamed it as `IsOccupied` defined in `transaction_manager.cpp`.